docs/src/tutorials/symbmc.md

```diff
@@ -0,0 +1,657 @@
+
+### Initial states
+The second component, `S0`, the set of all initial states, is derived from `S` and `Init`. 
+The initial state predicate is a Boolean formula, in which specification-variables appear as free logic variables. 
+One can see the operator `Init` as a predicate `S0 \in [S -> BOOLEAN]`, that means that, given a state `s \in S`, the formula obtained by replacing all occurrences of variable names `Ai` in `Init` with the values `s.Ai` (denoted `S0(s)`) is a Boolean formula with no free variables (in a well-typed, parseable specification), which evaluates to either `TRUE` or `FALSE`. By the subset-predicate equivalence, we treat the predicate `S0` as a subset of `S` (as it is equivalent to `{ s \in S: S0(s) = TRUE }`).
+
+For example, given
+```tla
+VARIABLE 
+  \* @type: Int;
+  x,
+  \* @type: Int;
+  y
+
+Init == x \in 3..5 /\ y = 2 
+```
+we see that `S = [x: Z, y: Z]` and `S0 = { [x |-> 3, y |-> 2], [x |-> 4, y |-> 2], [x |-> 5, y |-> 2] }`.
+
+### Transitions
+Similar to `S0`, `->` is derived from `S` and `Next`. If `S0` is a single-argument predicate `S0 \in [S -> BOOLEAN]`, then `->` is a relation `-> \in [S^2 -> BOOLEAN]`. 
+`->(s1,s2)` is the formula obtained by replacing all occurrences of variable names `Ai` in `Next` with the values `s1.Ai`, and all occurrences of `Ai'` with `s2.Ai`.
+By the same principle of subset-predicate equivalence, we can treat `->` as a subset of `S^2`.
+As mentioned in the notation section, it is generally more convenient to use the infix notation
+`s1 -> s2` over `->(s1, s2)`. We say that a state `s2` is a _successor_ of the state `s1` if `s1 -> s2`. 
+
+For example, given
+```tla
+VARIABLE 
+  \* @type: Int;
+  x,
+  \* @type: Int;
+  y
+
+Init == x \in 3..5 /\ y = 2 
+
+Next == x' \in { x, x + 1 } /\ UNCHANGED y
+```
+One can deduce, for any state `[x |-> a, y |-> b] \in S`, that it has two successors: `[x |-> a + 1, y |-> b]` and `[x |-> a, y |-> b]` because the following relations hold `[x |-> a, y |-> b]-> [x |-> a + 1, y |-> b]` and `[x |-> a, y |-> b] -> [x |-> a, y |-> b]`.
+
+### Reachable states
+Using the above definitions, we can define the set of states reachable in exactly `k`-steps, for `k \in N`, denoted by `R(k)`. We define `R(0) = S0` and for each `k \in N`,
+```tla
+R(k+1) := { t \in S: \E s \in R(k): s -> t }
+```
+
+Similarly, we can define the set of states reachable in _at most_ `k`-steps, denoted `r(k)`, for `k \in N` by
+```tla
+r(k) = R(0) \union R(1) \union ... \union R(k-1) \union R(k)
+```
+
+Finally, we define the et of all _reachable_ states, `R`, as the (infinite) union of all `r(k)`, over `k \in N`:
+```tla
+R = r(0) \union r(1) \union ...
```

Comment:
Usually, one uses `R*` for the transitive closure of `R`. You could write `C` or even `Closure`. We don't have a page limit here.

Reply:
While true, you're not looking for that, in general.
a) You'd need to define the term "transitive closure" for readers,
b) it'd be `->*`, and
c) you're only really interested in `{(a,_) \in ->* | a \in S0}`